Update machine settings information (#3772)
viamrobotics/docs: docs/manage/fleet/system-settings.md

To manage OS package updates using Viam, add an `"agent-syscfg"` object to the `"agent"` object in the machine's JSON configuration, if it doesn't already exist.
Then, add the `"upgrades"` field in its attributes:

```json
"agent": {
    "agent-syscfg": {
        "release_channel": "stable",
        "attributes": {
            "upgrades": {
                "type": "all|security|disabled"
            }
        }
    }
}
```

When the `upgrades` attribute is specified, `viam-agent` will install the `unattended-upgrades` package and replace `20auto-upgrades` and `50unattended-upgrades` in <FILE>/etc/apt/apt.conf.d/</FILE> with an Origins-Pattern list generated automatically from configured repositories on the system.
Custom repos installed on the system at the time the setting is enabled will be included.

You can set automatic upgrades for all packages by setting the field value to `{ "type": "all" }`.
Alternatively, you can set automatic upgrades for only packages containing `"security"` in their codename (for example `bookworm-security`), by setting the field value to `{ "type": "security" }`.

> To disable automatic upgrades, set the field value to `{ "type": "all" }`.

Reviewer: ```suggestion
To disable automatic upgrades, set the field value to `{ "type": "disabled" }`.
```

`viam-agent` will try to connect to `fallbackNetOne` first, since its priority is highest.
If the `fallbackNetOne` is not available or the machine can connect but internet is not available, `viam-agent` will then attempt to connect to `fallbackNetTwo`.

```json
"agent": {
    "agent-provisioning": {
        ...
        "attributes": {
            "roaming_mode": true,
            "networks": [
                {
                "type": "wifi",
                "ssid": "fallbackNetOne",
                "psk": "myFirstPassword",
                "priority": 30
                },
                {
                "type": "wifi",
                "ssid": "fallbackNetTwo",
                "psk": "mySecondPassword",
                "priority": 10
                }
            ]
        }
    }
}
```

For complete reference information, see [viam-agent](/configure/agent/#networks).

## Configure operating system logging

By default, the operating system defaults determine the disk space that `journald` uses to persist logs.
To adjust these settings update the `"agent"` value in the machine's JSON configuration.

> For complete reference information, see [viam-agent](/configure/agent/#agent-syscfg).

Author: ```suggestion
For complete reference information, see [viam-agent](/configure/agent/#agent-syscfg) and the [`journald` docs](https://www.freedesktop.org/software/systemd/man/latest/journald.conf.html#SystemMaxUse=).
```

---
linkTitle: "Configure machine settings"
title: "Configure machine operating system settings"
weight: 50
layout: "docs"
type: "docs"
description: "Configure network settings, operating system package updates and logging defaults."
---

The `viam-agent` configuration allows you to configure:

- [settings for package updates for the host operating system](#configure-os-package-updates)
- [networks a machine can connect to](#configure-networks)
- [parameters for operating system logging](#configure-operating-system-logging)

> ## Configure OS package updates

Reviewer 2: I'm unclear why a single field is explained in full three different times, with full json examples for each, when all that changes is one word. Wouldn't it be a lot easier to understand if the field was explained, and then just state that there are three options for "type"?

Custom repos installed on the system at the time the setting is enabled will be included.

To set automatic upgrades for all packages, add the `upgrades` field to the `agent-syscfg`'s `attributes` object as following.
You may need to add the `agent-syscfg` object to the `agent` object if it doesn't already exist.

```json
"agent": {
    "agent-syscfg": {
        "release_channel": "stable",
        "attributes": {
            "upgrades": {
                "type": "all"
            }
        }
    }
}
```

### Automatically upgrade security packages

When set, `viam-agent` will install the `unattended-upgrades` package and replace `20auto-upgrades` and `50unattended-upgrades` in <FILE>/etc/apt/apt.conf.d/</FILE> with an Origins-Pattern list generated automatically from configured repositories on the system.
Custom repos installed on the system at the time the setting is enabled will be included.

To set automatic upgrades for packages containing `"security"` in their codename (for example `bookworm-security`), add the `upgrades` field to the `agent-syscfg`'s `attributes` object as following.

> You may need to add the `agent-syscfg` object to the `agent` object if it doesn't already exist.

Reviewer 2: Why is this explanation repeated for every option here? Wouldn't it make more sense to just explain the overall purpose once at the top?

```json
"agent": {
    "agent-syscfg": {
        "release_channel": "stable",
        "attributes": {
            "upgrades": {
                "type": "security"
            }
        }
    }
}
```

### Disable automatic upgrades

When set, `viam-agent` will install the `unattended-upgrades` package and replace `20auto-upgrades` and `50unattended-upgrades` in <FILE>/etc/apt/apt.conf.d/</FILE> with an Origins-Pattern list generated automatically from configured repositories on the system.
Custom repos installed on the system at the time the setting is enabled will be included.

To set automatic upgrades, add the `upgrades` field to the `agent-syscfg`'s `attributes` object.
You may need to add the `agent-syscfg` object to the `agent` object if it doesn't already exist.

```json
"agent": {
    "agent-syscfg": {
        "release_channel": "stable",
        "attributes": {
            "upgrades": {

>                 "type": "security"

Reviewer 2: This should say "disabled" here, not "security" again.

{{< /alert >}}

The following configuration defines the connection information and credentials for two WiFi networks named `fallbackNetOne` and `fallbackNetTwo`.
`viam-agent` will try to connect to `fallbackNetOne` first, since its priority is highest.
If the `fallbackNetOne` is not available or the machine can connect but internet is not available, `viam-agent` will then attempt to connect to `fallbackNetTwo`.

```json
"agent": {
    "agent-provisioning": {
        ...
        "attributes": {
            "roaming_mode": true,
            "networks": [
                {
                "type": "wifi",
                "ssid": "fallbackNetOne",
                "psk": "myFirstPassword",
                "priority": 30
                },
                {
                "type": "wifi",
                "ssid": "fallbackNetTwo",
                "psk": "mySecondPassword",
                "priority": 10
                }
            ]
        }
    }
}
```

For complete reference information, see [viam-agent](/configure/agent/#networks).

> ### Example 2

Reviewer 2: Is this just a stray header?

Author: yes good point

> ### Example 2
> 
> 
> 

Author: ```suggestion
```

## Configure operating system logging

By default, the operating system defaults determine the disk space that `journald` uses to persist logs.
To adjust these settings update the `"agent"` value in the machine's JSON configuration.

For complete reference information, see [viam-agent](/configure/agent/#agent-syscfg).

> 

Reviewer 2: This should probably link to the docs for journald, which better explain what exactly these two fields mean. https://www.freedesktop.org/software/systemd/man/latest/journald.conf.html#SystemMaxUse=

> 

Reviewer 2: Should mention that if NOT set, both of these fields are set to 512MB by default (unless disable is set.)

Author: Does that mean this sentence is wrong or that the moment we set a logging value even if it is empty the value 512 is used? 

```
By default, the operating system defaults determine the disk space that `journald` uses to persist **logs.**
```

Reviewer 2: Correct. We enforce 512MB by default (per request from Eliot) when this is not set, which is typically smaller than most OS defaults. That's why I added the "disable" option to avoid this for people that don't want it changed.

- [networks a machine can connect to](#configure-networks)
- [parameters for operating system logging](#configure-operating-system-logging)

## Manage OS package updates

By default, the configuration in <FILE>/etc/apt/apt.conf.d/</FILE> determines the behavior for updating operating system packages.
To manage OS package updates using Viam, add an `"agent-syscfg"` object to the `"agent"` object in the machine's JSON configuration, if it doesn't already exist.
Then, add the `"upgrades"` field in its attributes:

```json
"agent": {
    "agent-syscfg": {
        "release_channel": "stable",
        "attributes": {
            "upgrades": {
                "type": "all|security|disabled"
            }
        }
    }
}
```

> When the `upgrades` attribute is specified, `viam-agent` will install the `unattended-upgrades` package and replace `20auto-upgrades` and `50unattended-upgrades` in <FILE>/etc/apt/apt.conf.d/</FILE> with an Origins-Pattern list generated automatically from configured repositories on the system.

Reviewer 2: ```suggestion
When the `type` attribute is set to `upgrades`, `viam-agent` will install the `unattended-upgrades` package and replace `20auto-upgrades` and `50unattended-upgrades` in <FILE>/etc/apt/apt.conf.d/</FILE> with an Origins-Pattern list generated automatically from configured repositories on the system.
```
[nit] `type` is the attribute. `upgrades` is the value of that attribute.

Author: what you're describing would mean `"type": "upgrades"`. I can say the type field is set for `"upgrades"`.

```suggestion
When the `type` attribute is specified for `"upgrades"`, `viam-agent` will install the `unattended-upgrades` package and replace `20auto-upgrades` and `50unattended-upgrades` in <FILE>/etc/apt/apt.conf.d/</FILE> with an Origins-Pattern list generated automatically from configured repositories on the system.
```

Custom repos installed on the system at the time the setting is enabled will be included.

You can set automatic upgrades for all packages by setting the field value to `{ "type": "all" }`.
Alternatively, you can set automatic upgrades for only packages containing `"security"` in their codename (for example `bookworm-security`), by setting the field value to `{ "type": "security" }`.

> To disable automatic upgrades, set the field value to `{ "type": "all" }`.

Reviewer 2: ```suggestion
To disable automatic upgrades, set the field value to `{ "type": "disabled" }`.
```
